src/main.rs

```diff
@@ -34,40 +40,41 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         // Read the file contents
         let file_contents = fs::read_to_string(entry.path())?;
 
-        // Skip files without TODOs
-        if !contains_todo_type(&file_contents) {
+        // Skip files without comments
+        if !file_contents.contains("//") && !file_contents.contains("<!--") {
             ok_files += 1;
             continue;
         }
 
+        // Check and parse todos
         for line in file_contents.lines().enumerate() {
-            if contains_todo_type(line.1) {
-                todos
-                    .entry(entry.path().display().to_string())
-                    .or_default()
-                    .push(line.into());
+            if contains_todo_type(line.1).is_ok() {
+                if needle.is_some() && !line.1.contains(needle.as_ref().unwrap()) {
+                    filtered_files += 1;
+                    continue;
+                }
+
+                let todo = (line.0, line.1, entry.path().display().to_string());
```

Comment:
Pass the filename when creating a `Todo` using `.into()`. This way, we can use a simple `Vec<Todo>` to store all the todos.